src/gleam/dynamic.gleam

```diff
@@ -290,6 +290,35 @@ pub fn result(
   }
 }
 
+fn do_try_map_with_index(
+  list: List(a),
+  fun: fn(a) -> Result(b, e),
+  acc: List(b),
+) -> Result(List(b), #(Int, e)) {
+  case list {
+    [] -> Ok(list.reverse(acc))
+    [x, ..xs] ->
+      case fun(x) {
+        Ok(y) -> do_try_map_with_index(xs, fun, [y, ..acc])
+        Error(error) -> Error(#(list.length(acc), error))
```

Comment:
Counting a list is expensive so let's hold on to the index instead.
```suggestion
        Ok(y) -> decode_list(xs, fun, index + 1, [y, ..acc])
        Error(error) -> Error(#(int.to_string(index), error))
```

Reply:
Hah, that's how I implemented it first, reverted back to using an accumulator :)